### ISSUE TITLE
kni module support time ?

### ISSUE THREAD

**Issue by awzhgw** [mtcp-stack/mtcp#259]
Hello, due to our network environment restrictions, the company's IP must use BGP to publish, after using MTCP and DPDK, BGP heartbeat packets will be blocked by DPDK, the system can not get the heartbeat, IP will be invalid.. I see MTCP The branch of kni_module supports KNI forwarding, so how do I use this branch to forward the BGP packet protocol? Will this branch be merged into the main branch?

**Comment by ajamshed** [comment]
@awzhgw,

I started working on the KNI extension some time back, but realized that using eth_vdev+af_packet mode will be a far better option than KNI. I will try to update the stack some time the following week and will let you know once this is patched up.
